tests/integration/bond_test.py

```diff
@@ -262,3 +262,46 @@ def test_set_bond_mac_address(eth1_up):
         current_state = statelib.show_only(('bond99',))
         bond99_cur_state = current_state[INTERFACES][0]
         assert bond99_cur_state[Interface.MAC] == MAC1.upper()
+
+
+@pytest.fixture
+def bond0():
+    with bond_interface(name='bond99', slaves=[]) as bond_state:
+        yield bond_state
+
+
+def assert_mac_address_behavior(bond0):
+    slaves_state = statelib.show_only(('bond99', 'eth1', 'eth2'))
+    bond99_slaves_state = slaves_state[Interface.KEY][0]
+    eth1_slaves_state = slaves_state[Interface.KEY][1]
+    eth2_slaves_state = slaves_state[Interface.KEY][2]
+
+    bond99_slaves_mac = bond99_slaves_state[Interface.MAC]
+    eth1_slaves_mac = eth1_slaves_state[Interface.MAC]
+    eth2_slaves_mac = eth2_slaves_state[Interface.MAC]
+
+    assert (bond99_slaves_mac ==
+            eth1_slaves_mac ==
+            eth2_slaves_mac)
+
+
+def test_bond_with_no_slaves_have_same_mac(bond0):
+    assert_mac_address_behavior(bond0)
+
+
+def test_bond_with_two_slaves_has_same_mac(bond0):
+    bond0[INTERFACES][0]['link-aggregation']['slaves'] = ['eth1', 'eth2']
+    netapplier.apply(bond0)
+    assert_mac_address_behavior(bond0)
+
+
+def test_bond_with_all_removed_slaves_has_same_mac(bond0):
```

Comment:
If all slaves are removed, why should they have the same MAC address?

Reply:
you need to assert that the slavees and the bond all have different MAC addresses here.